yogstation/code/modules/guardian/abilities/special/pocket.dm

```diff
@@ -0,0 +1,387 @@
+		if(errorList.len)	// reebe failed to load
+			message_admins("A pocket dimension failed to load!")
+			log_game("A pocket dimension failed to load!")
+			return FALSE
+		for(var/datum/parsed_map/PM in pocket_dim)
+			PM.initTemplateBounds()
+		guardian.pocket_dim = "Pocket Dimension [GLOB.pocket_dim++]"
+		GLOB.pocket_mirrors[guardian.pocket_dim] = list()
+		var/pz = get_pocket_z()
+		eye = new
+		eye.guardian = guardian
+		eye.pocket_z = pz
+		eye.PDS = src
+		eye.name = "Inactive Guardian Eye ([guardian])"
+		for(var/turf/open/indestructible/pocketspace/PS in world)
+			if(PS.z == pz)
+				GLOB.pocket_mirrors[guardian.pocket_dim] += PS
+
+/datum/guardian_ability/major/special/pocket/Remove()
+	guardian.RemoveSpell(PD)
+	guardian.RemoveSpell(PDM)
+
+/datum/guardian_ability/major/special/pocket/proc/get_pocket_z()
+	if(!guardian.pocket_dim)
+		return
+	var/list/zs = SSmapping.levels_by_trait(guardian.pocket_dim)
+	if(!LAZYLEN(zs))
+		return null
+	return zs[1]
+
+/datum/guardian_ability/major/special/pocket/proc/manifest_dimension(pos_already_set = FALSE)
+	var/pocket_z = get_pocket_z()
+	if(!pocket_z)
+		return
+	destroy_pocket_mirror(pocket_z)
+	manifesting = TRUE
+	if(LAZYLEN(GLOB.pocket_mirrors[guardian.pocket_dim]))
+		for(var/turf/open/indestructible/pocketspace/PS in GLOB.pocket_mirrors[guardian.pocket_dim])
+			PS.vis_contents.Cut()
+	var/corrected_max = manifested_at_x
+	var/corrected_may = manifested_at_y
+	if(!pos_already_set) // don't subtract 4 if the pos is set, because we will have already subtracted by 4.
+		manifested_at_x = CLAMP(guardian.x - 4, 1, world.maxx)
+		corrected_max = manifested_at_x
+		manifested_at_y = CLAMP(guardian.y - 4, 1, world.maxy)
+		corrected_may = manifested_at_y
+		manifested_at_z = get_final_z(guardian)
+	for(var/mx = 0 to 8)
+		for(var/my = 0 to 8)
+			var/turf/us = locate(corrected_max + mx, corrected_may + my, manifested_at_z)
+			var/turf/them = locate(mx + 1, my + 1, pocket_z)
+			if(us && them)
+				var/obj/effect/manifestation/M = new(us)
+				M.vis_contents += them
+				manifestations += M
+	addtimer(VARSET_CALLBACK(src, manifesting, FALSE), 3 SECONDS)
+
+/datum/guardian_ability/major/special/pocket/proc/demanifest_dimension()
+	manifesting = TRUE
+	for(var/obj/effect/manifestation/M in manifestations)
+		manifestations -= M
+		animate(M, alpha = 0, time = 3 SECONDS, easing = LINEAR_EASING)
+		QDEL_IN(M, 3 SECONDS)
+	var/pocket_z = get_pocket_z()
+	if(pocket_z)
+		addtimer(CALLBACK(GLOBAL_PROC, .proc/update_pocket_mirror, pocket_z, manifested_at_x, manifested_at_y, manifested_at_z), 3.5 SECONDS)
+	addtimer(VARSET_CALLBACK(src, manifesting, FALSE), 3 SECONDS)
+
+/datum/guardian_ability/major/special/pocket/proc/check_if_teleport(mob/living/L)
+	var/list/pocket_z = get_pocket_z()
+	if(!pocket_z)
+		return
+	if(get_final_z(L) != pocket_z)
+		take_effects(L)
+
+/datum/guardian_ability/major/special/pocket/proc/add_effects(mob/living/L)
+	L.status_flags |= GODMODE
```

Comment:
Did you port my change that allows you to actually heal in godmode?

Reply:
ported the healing in godmode bit